astropy/coordinates/baseframe.py

```diff
@@ -811,8 +814,23 @@ def represent_as(self, base, s='base', in_frame_units=False):
                                       for comp in diff.components)
                     for comp, new_attr_unit in zip(diff.components,
                                                    new_attrs['units']):
-                        if (new_attr_unit and
-                                hasattr(self._data.differentials['s'], comp)):
+                        # Some special-casing to treat a situation where the
```

Comment:
Other possibilities: At the frame level, if a non-existing component is requested, 
* return `None` for that component
* return `0 dimensionless`